xGame/Java/xgame/src/main/java/com/xgame/service/ChessMatchService.java

```diff
@@ -216,4 +216,50 @@ else if(match.getBlackPlayer().getId() == playerId) {
 		var updatedMatch = matchRepo.save(match);
 		return updatedMatch.getMatchOutcome();
 	}
+	
+	/**
+	 * Denies a draw proposed by a given player's opponent. Undo opponent suggestion, match remains ongoing.
+	 * @param matchId - Id of match where a draw is being denied
+	 * @param playerId - player denying the draw
+	 * @return Status of match
+	 */
+	@Override
+	public MatchStatus denyDraw(int matchId, int playerId) {
+		var m = matchRepo.findById(matchId);
+		m.orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "No match with that ID exists."));
+		var u = userRepo.findById(playerId);
+		u.orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "No user with that ID exists."));
+		
+		var match = m.get();
+		var player = u.get();
+		var opponentId = playerId == match.getWhitePlayer().getId() ? 
+				match.getBlackPlayer().getId() : 
+				match.getWhitePlayer().getId();
+		var opponent = userRepo.findById(opponentId);
+		
+		if(match.getMatchStatus() != MatchStatus.INPROGRESS) {
+			throw new ResponseStatusException(
+					HttpStatus.BAD_REQUEST, "Cannot propose draw for match " + matchId + " because it is not in progress.");
+		}
```

Comment:
move this to right after the `m.orElseThorw()` so you don't make extra calls to the database getting users.